### PR TITLE
[58] Develop an E-commerce Store Database 

### PR DESCRIPTION
**Pull request by MYKEL7**
## Task: `Develop an E-commerce Store Database `
- `Created an e-commerce database for Mix Stores`

### CODE REVIEW

File: submissions/onyemeMichael/database/init.sql [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+-- Insert into addresse table
+INSERT INTO addresses (street_adress, city, state_name, zipcode, country, phone_number, customer_id)
+VALUES ('Mykel str', 'Apapa', 'Lagos', 100335, 'Nigeria', 08182515388, 1 ),
+       ('Chuks str', 'Yaba', 'Lagos', 100375, 'Nigeria', 07012345678, 2 );
+
+
+-- Order Table
+CREATE TABLE orders (
+	id INT NOT NULL AUTO_INCREMENT,
+    order_amount DECIMAL (20,4) NOT NULL,
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:99]: Are you sure we have up to 20? same goes for others

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/onyemeMichael/database/init.sql [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+       ('Emmanuel', 'Ojo', 'emmaneul@email.com', 'emmanuel', '07031803086'),
+       ('Ada', 'Mgbede', 'adaamgbede@email.com', 'adaa', '08109030672'),
+       ('Toluwalase', 'Akintoye', 'toluakintoye@email.com', 'tolu', '08012785678');
+       
+-- Products table
+CREATE TABLE products (
+	id INT NOT NULL AUTO_INCREMENT,
+    product_name VARCHAR (50) NOT NULL,
+    product_description VARCHAR (255) NOT NULL,
+    product_image BLOB,
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:45]: I think should be NOT NULL as well

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by MYKEL7** [comment]
Done

File: submissions/onyemeMichael/database/init.sql [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+VALUES ('Paul', 'Onyeme', 'paulonyeme@email.com', 'paul'),
+	   ('Mayowa', 'Ojo', 'mayawaojo@email.com', 'mayowa');
+
+-- Rigistered Customers Addresses
+CREATE TABLE addresses (
+	id INT NOT NULL AUTO_INCREMENT,
+    street_adress VARCHAR (255) NOT NULL,
+    city VARCHAR (50) NOT NULL,
+    state_name VARCHAR (50) NOT NULL,
+    zipcode INT NOT NULL,
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:82]: Read the question again concerning `zipcode`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/onyemeMichael/database/init.sql [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    country VARCHAR (50) NOT NULL,
+    phone_number VARCHAR (20) NOT NULL,
+    customer_id INT NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (customer_id) REFERENCES customers(id) ON DELETE CASCADE
+);
+
+-- Insert into addresse table
+INSERT INTO addresses (street_adress, city, state_name, zipcode, country, phone_number, customer_id)
+VALUES ('Mykel str', 'Apapa', 'Lagos', 100335, 'Nigeria', 08182515388, 1 ),
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:92]: phone_number should a string

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Comment by MYKEL7** [comment]
Done

File: submissions/onyemeMichael/database/init.sql [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    country VARCHAR (50) NOT NULL,
+    phone_number VARCHAR (20) NOT NULL,
+    customer_id INT NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (customer_id) REFERENCES customers(id) ON DELETE CASCADE
+);
+
+-- Insert into addresse table
+INSERT INTO addresses (street_adress, city, state_name, zipcode, country, phone_number, customer_id)
+VALUES ('Mykel str', 'Apapa', 'Lagos', '100335', 'Nigeria', '08182515388', '1' ),
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:92]: customer_id is an `INT` and not a string

Comment (adaugochi) [reply submissions/onyemeMichael/database/init.sql:92]: Same as zip code. Please note the differences

**Review by adaugochi — CHANGES_REQUESTED** [review]
Kindly ensure you rerun your query when correction is made. Delete the DB and rerun the query again

File: submissions/onyemeMichael/database/init.sql [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    product_id INT NOT NULL,
+    order_id INT NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (product_id) REFERENCES products(id) ON DELETE CASCADE,
+    FOREIGN KEY (order_id) REFERENCES orders(id) ON DELETE CASCADE
+);
+
+-- Insert into order items table
+INSERT INTO order_items (quantity, unit_price, total_amount, product_id, order_id)
+VALUES ('1', '138999', '138999', '1', '1');
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:126]: unit_price and amount are DECIMAL, Why are you inserting a string

File: submissions/onyemeMichael/database/init.sql [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    order_amount DECIMAL (16,4) NOT NULL,
+    ordered_at DATETIME NOT NULL,
+    customer_id INT NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (customer_id) REFERENCES customers(id) ON DELETE CASCADE
+);
+
+-- Insert into orders table
+INSERT INTO orders (order_amount, ordered_at, customer_id)
+VALUES ('138999', now(), '1'),
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:108]: `order_amount` is a DECIMAL, but you are inserting string

**Review by adaugochi — CHANGES_REQUESTED** [review]
Kindly note this differences

File: submissions/onyemeMichael/database/init.sql [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    product_id INT NOT NULL,
+    order_id INT NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (product_id) REFERENCES products(id) ON DELETE CASCADE,
+    FOREIGN KEY (order_id) REFERENCES orders(id) ON DELETE CASCADE
+);
+
+-- Insert into order items table
+INSERT INTO order_items (quantity, unit_price, total_amount, product_id, order_id)
+VALUES (1, 138999, 138999, 1, 1);
```
Comment (adaugochi) [inline submissions/onyemeMichael/database/init.sql:126]: amount and price are `decimal` not `int`

Comment (adaugochi) [reply submissions/onyemeMichael/database/init.sql:126]: Same for others too

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by adaugochi — APPROVED** [review]

**Pull request merged**